tools/intellij-plugin/src/creator/steps/ValidationUtil.kt

```diff
@@ -0,0 +1,128 @@
+        companion object {
+            init {
+                registerValidator<Pattern> { annotation, component ->
+                    if (component.text?.matches(Regex(annotation.value)) != true) {
+                        report("请正确填写 ${annotation.tipName}")
+                    }
+                }
+            }
+        }
+    }
+
+    fun interface Validator<in A : Annotation> {
+        @Throws(ValidationException::class)
+        fun ValidationContext.validate(annotation: A, component: JTextComponent)
+
+        @Throws(ValidationException::class)
+        fun validate(annotation: A, component: JTextComponent) {
+            ValidationContext.run { validate(annotation, component) }
+        }
+
+        object ValidationContext {
+            fun report(message: String): Nothing = throw ValidationException(message)
+        }
+    }
+
+    class ValidationException(message: String) : Exception(message)
+
+    companion object {
+        private data class RegisteredValidator<A : Annotation>(val type: KClass<A>, val validator: Validator<A>)
+
+        private val validators: MutableCollection<RegisteredValidator<*>> = ConcurrentLinkedQueue()
+
+        private inline fun <reified A : Annotation> registerValidator(validator: Validator<A>) {
+            validators.add(RegisteredValidator(A::class, validator))
+        }
+
+        fun popup(message: String, component: JComponent) {
+            JBPopupFactory.getInstance()
+                .createHtmlTextBalloonBuilder(message, MessageType.ERROR, null)
+                .setFadeoutTime(2000)
+                .createBalloon()
+                .show(RelativePoint.getSouthWestOf(component), Balloon.Position.below)
+        }
+
+        /**
+         * @return `true` if no error
+         */
+        fun doValidation(step: ModuleWizardStep): Boolean {
+            fun validateProperty(field: Field): Boolean {
+                field.isAccessible = true
+                val annotationsToValidate =
+                    validators.associateBy { (type: KClass<out Annotation>) ->
+                        field.annotations.find { it::class == type }
+                    }
+
+                for ((annotation, validator) in annotationsToValidate) {
+                    if (annotation == null) continue
+                    val component = field.get(step) as JTextComponent
+                    try {
+                        validator.validator.cast<Validator<Annotation>>().validate(annotation, component)
+                    } catch (e: ValidationException) {
+                        popup(e.message ?: e.toString(), component)
+                        return false // report one error only
+                    }
+                }
+                return true
+            }
+
+            for (prop in step::class.java.declaredFields) {
+                if (!validateProperty(prop)) return false
+            }
+            return true
```

Comment:
这个逻辑是不是不会把全部的错误都显示出来, 还是有什么别的原因
```suggestion
            var result = true
            for (prop in step::class.java.declaredFields) {
                if (!validateProperty(prop)) result = false
            }
            return result
```

Reply:
intentionally 只显示一个错误